src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -1393,34 +1401,33 @@ bool SystemDictionaryShared::should_be_excluded(InstanceKlass* k) {
   }
 
   if (k->has_old_class_version() && k->is_linked()) {
-    warn_excluded(k, "Old class has been linked");
-    return true;
+    return warn_excluded(k, "Old class has been linked");
   }
 
-  InstanceKlass* super = k->java_super();
-  if (super != NULL && should_be_excluded(super)) {
+  if (k->is_hidden() && !is_registered_lambda_proxy_class(k)) {
     ResourceMark rm;
-    log_warning(cds)("Skipping %s: super class %s is excluded", k->name()->as_C_string(), super->name()->as_C_string());
+    log_debug(cds)("Skipping %s: %s", k->name()->as_C_string(), "Hidden class");
     return true;
   }
 
-  if (k->is_hidden() && !is_registered_lambda_proxy_class(k)) {
+  InstanceKlass* super = k->java_super();
+  if (super != NULL && check_for_exclusion(super, NULL)) {
     ResourceMark rm;
-    log_debug(cds)("Skipping %s: %s", k->name()->as_C_string(), "Hidden class");
+    log_warning(cds)("Skipping %s: super class %s is excluded @%p", k->name()->as_C_string(), super->name()->as_C_string(), super);
```

Comment:
Pretty sure we still don't/can't use %p in shared code, but should be INTPTR_FORMAT and use p2i(super)

Reply:
I reverted back to the original code and removed the %p. I added %p for temporary debugging only.